<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <!-- dummy detection and sensor data -->
  <arg name="initial_engage_state" default="true"/>

  <group>
    <push-ros-namespace namespace="simulation"/>

    <node pkg="npc_simulator" exec="npc_simulator_main" name="npc_simulator" output="screen">
      <param name="/initial_engage_state" value="$(var initial_engage_state)"/>
      <remap from="/input/engage" to="npc_simulator/engage"/>
      <remap from="/input/object" to="npc_simulator/object_info"/>
      <remap from="/input/vector_map" to="/map/vector_map"/>
      <remap from="/input/ego_vehicle_pose" to="/current_pose"/>
      <remap from="/output/dynamic_object_info" to="/simulation/dummy_perception_publisher/object_info"/>
      <remap from="/output/debug_object_info" to="npc_simulator/ground_truth_object_info"/>
    </node>
  </group>
</launch>
